### PR TITLE
assimp: add v5.4.1

### PR DESCRIPTION
**Pull request by valgur**
https://github.com/assimp/assimp/releases/tag/v5.4.0

No new formats or exporters were added in this release.

### CODE REVIEW

**Review by EstebanDugueperoux2 — APPROVED** [review]

**Comment by RubenRBS** [comment]
Should we ensure we don't get bitten by changes like https://github.com/assimp/assimp/compare/v5.3.1...v5.4.0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR140 in the future? `ASSIMP_BUILD_ZLIB` changed value but we don't set it in the generate method

**Comment by valgur** [comment]
@RubenRBS Thanks for catching that. I added explicit handling of all relevant CMake options.

**Comment by sunbubble** [comment]
https://github.com/assimp/assimp/releases/tag/v5.4.1

5.4.1 has just been released and addresses some Draco relating issues. Could it be related to the warnings emitted above?

**Comment by sunbubble** [comment]
@valgur thanks for bumping this so fast. Out of interest, do you know whether the warnings in the build is what's holding up the merging of this PR? Or is just the queue taking its time?

**Comment by valgur** [comment]
@sunbubble The queue of open PRs is simply very long (https://github.com/ericLemanissier/conan-center-index-pending-review/issues/3), although simple PRs like this one usually get reviewed faster.
I think the Conan devs are currently focusing on a major overhaul of the CCI CI system and Conan itself, so reviews can be somewhat slow.

File: recipes/assimp/5.x/conanfile.py [hunk 1]
```diff
@@ -298,6 +318,8 @@ def _patch_sources(self):
         ]:
             save(self, os.path.join(self.source_folder, "contrib", contrib_header),
                  f"#include <{include}>\n")
+        if Version(self.version) >= "5.4.0":
+            rmdir(self, self.source_path.joinpath("contrib", "utf8cpp"))
```
Comment (RubenRBS) [inline recipes/assimp/5.x/conanfile.py:322]: This seems to only be used in samples and there's a change in the release mentioning this, but why is it only now removed? (I might have checked the wrong diff now that I think about it!)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot @valgur :)
Minor question, otherwise looks good

File: recipes/assimp/5.x/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/assimp/5.x/conanfile.py:322]: Oh, it's probably unnecessary in that case then. Let me see if it builds ok without this...

Comment (valgur) [reply recipes/assimp/5.x/conanfile.py:322]: It was necessary. :slightly_smiling_face:

Comment (RubenRBS) [reply recipes/assimp/5.x/conanfile.py:322]: Thanks for double-checking :)

**Comment by RubenRBS** [comment]
I'm seeing some `assimp/5.4.1: WARN: replace_in_file didn't find pattern '-fPIC' in '/Users/jenkins/workspace/prod-v2/bsr/38132/eabea/p/b/assimcb1569f627615/b/src/code/CMakeLists.txt' file.` in https://c3i.jfrog.io/c3i/misc-v2/logs/pr/23735/3-macos-m1-clang/assimp/5.4.1//76e545471fe165ddd092b4cbb0361627c0f28db3-build.txt but it seems it's still there! https://github.com/assimp/assimp/blob/942518a4ff148ec10c62c425825c8186eec7fc65/CMakeLists.txt#L311

Are we thus not patching things properly?

**Comment by valgur** [comment]
@RubenRBS The logic was correct, but I cleaned up the patching to avoid the use of `strict=False` nevertheless.

**Review by RubenRBS — APPROVED** [review]
Thanks!

**Review by toge — APPROVED** [review]

**Pull request merged**